modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/test/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/util/DDMFormInstanceStatusUtilTest.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.form.web.internal.display.context.util;
+
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstanceSettings;
+
+import java.util.TimeZone;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+/**
+ * @author Rodrigo Paulino
+ */
+@RunWith(PowerMockRunner.class)
+public class DDMFormInstanceStatusUtilTest extends PowerMockito {
+
+	@Test
+	public void testFormExpired() throws Exception {
+		DDMFormInstanceSettings ddmFormInstanceSettings =
+			_mockDDMFormInstanceSettings("1987-09-22", false);
+
+		Assert.assertTrue(
+			DDMFormInstanceStatusUtil.isFormExpired(
+				_mockDDMFormInstance(ddmFormInstanceSettings), null));
+		Assert.assertTrue(
+			DDMFormInstanceStatusUtil.isFormExpired(
+				_mockDDMFormInstance(ddmFormInstanceSettings),
+				TimeZone.getDefault()));
+	}
+
+	@Test
+	public void testFormNotExpiredWithInexistentForm() throws Exception {
+		Assert.assertFalse(DDMFormInstanceStatusUtil.isFormExpired(null, null));
+	}
+
+	@Test
+	public void testFormNotExpiredWithNeverExpireSetting() throws Exception {
+		DDMFormInstanceSettings ddmFormInstanceSettings =
+			_mockDDMFormInstanceSettings(null, true);
+
+		Assert.assertFalse(
+			DDMFormInstanceStatusUtil.isFormExpired(
+				_mockDDMFormInstance(ddmFormInstanceSettings), null));
+	}
+
+	private DDMFormInstance _mockDDMFormInstance(
+			DDMFormInstanceSettings ddmFormInstanceSettings)
+		throws Exception {
+
+		DDMFormInstance ddmFormInstance = mock(DDMFormInstance.class);
+
+		when(
+			ddmFormInstance.getSettingsModel()
+		).thenReturn(
+			ddmFormInstanceSettings
+		);
+
+		return ddmFormInstance;
+	}
+
+	private DDMFormInstanceSettings _mockDDMFormInstanceSettings(
+		String expirationDate, Boolean neverExpire) {
+
+		DDMFormInstanceSettings ddmFormInstanceSettings = mock(
+			DDMFormInstanceSettings.class);
+
+		if (expirationDate != null) {
```

Comment:
@rodrigopaulino is this `if` really necessary?

Reply:
Yeah, not really. It was just for future usages in case some test needed it to not mock the expiration date.

Reply:
It was just out of curiosity. I don't see any problem keeping it. Thanks!